### PR TITLE
fix: Bad formatting of values in the table and duplicate formatting methods

### LINKED ISSUE #6: Amount component
hello, I have modified the amount component, now it consults in a .JSON file the default variables for this component, also when listening to the onblur event it validates the text input which is not allowed in this component.
file:///home/erp-04/Pictures/Screenshot_20181226_123438.png

### LINKED ISSUE #7: Support to integer component
Hello, it is necessary to migrate the entire component, this will support data imported from gRPC and must conform to ADempiere standards. It should not allow decimal values.

**ADempiere**
zk-integer

[**Element-UI**](https://element.eleme.io/#/en-US/component/input-number#basic-usage)
element-integer

**ADempiere-Vue Desktop**
adempiere-vue-integer

**ADempiere-Vue Mobile**
adempiere-vue-mobile-integer

### LINKED ISSUE #10: Support to numbers component

### LINKED ISSUE #17: Suport to Costs-Prices component
Migration of the costs-prices component for better performance.

### LINKED ISSUE #63: Support to ID Component
hello you need to do the ID migration so that it is maintained it is important to say that this component is based on vue.js and element ui

### LINKED ISSUE #244: [Bug Report] Show totals without amounts.
#### Steps to reproduce

1. Open _**Views.**_ window.
2. In the context menu of records table, select _Show Totals_.
3. At the bottom of the table, a summation cell appears, but there are no amounts to add.

#### Screenshot or Gif（截图或动态图）
show-totals

#### Expected behavior
If there is no cell with amounts in the record table, the option to show totals should not appear.

#### Other relevant information
- Your OS: Kubuntu 19.04
- Node.js version: 12.13.1

### PR DESCRIPTION
**Pull request by EdwinBetanc0urt**
## Bug report / Feature

Some duplicated functions are removed.

Formatting of number display in cells as well as table summation is wrong.

Moved functions that deal with number formatting to `src/utils/ADempiere/numberFormat.js` file.

Created a method for formatting numeric fields using the diplayType implementing the existing `formatCurrency` and `formatQuant` methods.

Method for number formatting is used in the `formatField` and number field components.

#### Steps to reproduce
Display numerical values in the: 
- Data tables, Amount, Number, Price+Cost and Quantity columns if it has no values then in the decimals it formats it as an integer when it should show the decimals in zero. If it has no value it should show 0 and not empty for numeric fields in the table.
- FieldNumber component.

#### Screenshot or Gif

Before this PR:
https://user-images.githubusercontent.com/20288327/123015447-67811500-d396-11eb-8d85-385901943dd6.mp4

After this PR:
https://user-images.githubusercontent.com/20288327/122630043-0e13a000-d08f-11eb-8dc9-6850fedc3e63.mp4

#### Other relevant information
* Your OS: Linux Mint 19.1 Cinnamon x64. 
* Browser: Mozilla Firefox 88.0.1.
* Node.js version: 14.17.0.
* NPM version: 7.17.0.
* adempiere-vue version: 4.3.1.

#### Additional context
closes https://github.com/adempiere/adempiere-vue/issues/6, closes https://github.com/adempiere/adempiere-vue/issues/7, closes https://github.com/adempiere/adempiere-vue/issues/10, closes https://github.com/adempiere/adempiere-vue/issues/17, closes https://github.com/adempiere/adempiere-vue/issues/63, fixes https://github.com/adempiere/adempiere-vue/issues/244

### CODE REVIEW

**Comment by yamelsenih** [comment]
This is a duplicated funcionality, please reconsider it. Are too many files without changes here

File: src/utils/ADempiere/numberFormat.js [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+  }
+
+  // const precision = getStandardPrecision()
+
+  if (isEmptyValue(country)) {
+    country = getCountryCode()
+  }
+
+  // get formatted currency number
+  return new Intl.NumberFormat(country, {
```
Comment (yamelsenih) [inline src/utils/ADempiere/numberFormat.js]: This already exist for valueFormat

**Review by yamelsenih — CHANGES_REQUESTED** [review]

File: src/utils/ADempiere/numberFormat.js (hunk 1 above)
Comment (EdwinBetanc0urt) [reply src/utils/ADempiere/numberFormat.js]: this was moved to numberFormat to place there all the methods and helpers that have to do with numbers.

Delete https://github.com/adempiere/adempiere-vue/pull/932/files#diff-f8797464a94c1bcbfc855b09cbe72e05ba39484ccbbb0dc3cf42a0db83c89093L165

Add https://github.com/adempiere/adempiere-vue/pull/932/files#diff-f0b105e877af046f431bd18c228f85a91c1215a613551754a261fe981fce9fedR82

File: src/utils/ADempiere/valueFormat.js [hunk 2]
```diff
@@ -161,97 +163,135 @@ export function formatDate(date, isTime = false) {
...
+  if (isEmptyValue(currency)) {
     currency = getCurrency()
   }
-  //  Get formatted number
-  return new Intl.NumberFormat(getCountryCode(), {
+
+  // const precision = getStandardPrecision()
+
+  if (isEmptyValue(country)) {
+    country = getCountryCode()
```
Comment (yamelsenih) [inline src/utils/ADempiere/valueFormat.js:179]: Don't touch it if not exists changes

File: src/utils/ADempiere/valueFormat.js [hunk 3]
```diff
@@ -161,97 +163,135 @@ export function formatDate(date, isTime = false) {
...
+  }
+
+  // get formatted currency number
+  return new Intl.NumberFormat(country, {
     style: 'currency',
-    currency
-  }).format(number)
+    currency,
+    useGrouping: true,
+    // minimumFractionDigits: precision,
```
Comment (yamelsenih) [inline src/utils/ADempiere/valueFormat.js:187]: Don't touch it if not exists changes

File: src/utils/ADempiere/valueFormat.js [hunk 4]
```diff
@@ -161,97 +163,135 @@ export function formatDate(date, isTime = false) {
...
-export function formatQuantity(number) {
-  if (this.isEmptyValue(number)) {
-    return undefined
+/**
+ * Get formatted number, integer and decimal
+ * @param {number} value
+ * @returns {string} number format with thousands separator, precision
+ * @returns {boolean} number format without precision
+ */
+export function formatQuantity(value, isInteger = false) {
```
Comment (yamelsenih) [inline src/utils/ADempiere/valueFormat.js]: Don't touch it if not exists changes

File: src/utils/ADempiere/valueFormat.js [hunk 5]
```diff
@@ -161,97 +163,135 @@ export function formatDate(date, isTime = false) {
...
+  if (isInteger) {
+    // if (Number.isInteger(value)) {
+    precision = 0
   }
-  return Number.parseFloat(number).toFixed(2)
-  //  Get formatted number
+
+  // get formatted decimal number
+  return new Intl.NumberFormat(undefined, {
+    useGrouping: true, // thousands separator
```
Comment (yamelsenih) [inline src/utils/ADempiere/valueFormat.js:208]: Don't touch it if not exists changes

File: src/utils/ADempiere/numberFormat.js [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+/**
+ * Convert exponetial to decimals quantity
+ * @param {number} value ej: 314e-2, 4e+3
+ * @returns ej: 3.14, 400
+ */
+export function formatExponential(value) {
+  const regExpr = /(\d{1,})(e-)/g // number end e-
+  const strValue = value.toString()
+
+  if (regExpr.test(strValue)) {
```
Comment (yamelsenih) [inline src/utils/ADempiere/numberFormat.js:106]: Add it to value format instead

File: src/store/modules/ADempiere/system.js [hunk 7]
```diff
@@ -75,17 +91,20 @@ const system = {
...
+    getCurrency: (state, getter) => {
+      const { currencyIsoCode } = state.systemDefinition
 
-      return {
-        standardPrecision: standardPrecision || 2,
-        iSOCode: currencyIsoCode || 'USD'
-      }
+      return currencyIsoCode || 'USD'
+    },
+    getStandardPrecision: (state) => {
```
Comment (yamelsenih) [inline src/store/modules/ADempiere/system.js:111]: Don't touch it if not exists changes

File: src/components/ADempiere/DataTable/dataTables-Script.js [hunk 8]
```diff
@@ -723,34 +689,24 @@ export default {
...
-            }
-            return prev
-          }, 0)
-          sums[index] = this.formatNumber({
-            displayType,
-            number: total
-          })
-        }
+
+        const values = selectionData.map(item => Number(item[columnItem.property]))
```
Comment (yamelsenih) [inline src/components/ADempiere/DataTable/dataTables-Script.js:693]: Don't touch it if not exists changes

**Review by yamelsenih — CHANGES_REQUESTED** [review]

File: src/utils/ADempiere/valueFormat.js (hunk 5 above)
Comment (EdwinBetanc0urt) [reply src/utils/ADempiere/valueFormat.js:208]: The thousands separator of the format is not being taken.

File: src/utils/ADempiere/valueFormat.js (hunk 3 above)
Comment (EdwinBetanc0urt) [reply src/utils/ADempiere/valueFormat.js:187]: The thousands separator of the format is not being taken.

File: src/utils/ADempiere/numberFormat.js (hunk 6 above)
Comment (EdwinBetanc0urt) [reply src/utils/ADempiere/numberFormat.js:106]: I consider that it is better to have the methods in relation to the numbers in formatNumber.js.

File: src/utils/ADempiere/valueFormat.js (hunk 2 above)
Comment (EdwinBetanc0urt) [reply src/utils/ADempiere/valueFormat.js:179]: In some cases the country may change and the parameter can be used for this, if it is empty as it is by default the country code of the session is brought in.

File: src/store/modules/ADempiere/system.js (hunk 7 above)
Comment (EdwinBetanc0urt) [reply src/store/modules/ADempiere/system.js:111]: The change was reverted, although the getter of getCurrency (which returns the standard precision and the currency code) is not used, only new functionalities were added.

File: src/components/ADempiere/DataTable/dataTables-Script.js (hunk 8 above)
Comment (EdwinBetanc0urt) [reply src/components/ADempiere/DataTable/dataTables-Script.js:693]: The change is around the format of the summation totals table.

**Pull request closed without merging**